docs/data/joy/components/input/InputSubscription.tsx

```diff
@@ -0,0 +1,72 @@
+import * as React from 'react';
+import FormControl from '@mui/joy/FormControl';
+import FormLabel from '@mui/joy/FormLabel';
+import FormHelperText from '@mui/joy/FormHelperText';
+import Input from '@mui/joy/Input';
+import Button from '@mui/joy/Button';
+
+export default function InputSubscription() {
+  const [form, setForm] = React.useState<{
+    email: string;
+    status: 'initial' | 'loading' | 'failure' | 'sent';
+  }>({
+    email: '',
+    status: 'initial',
+  });
+
+  const handleSubmit = () => {
+    setForm((current) => ({ ...current, status: 'loading' }));
+    try {
+      // Replace timeout with real backend operation
+      setTimeout(() => {
+        setForm({ email: '', status: 'sent' });
+      }, 1500);
+    } catch (error) {
+      setForm((current) => ({ ...current, status: 'failure' }));
+    }
+  };
+
+  return (
+    <FormControl>
+      <FormLabel
+        sx={(theme) => ({
+          '--FormLabel-color': theme.vars.palette.primary[400],
+        })}
+      >
+        MUI Newsletter
+      </FormLabel>
+      <Input
+        sx={{ pt: 0.5, pb: 0.5 }}
+        placeholder="mail@mui.com"
+        type="email"
+        value={form.email}
+        onChange={(event) =>
+          setForm({ email: event.target.value, status: 'initial' })
+        }
+        error={form.status === 'failure'}
+        endDecorator={
+          <Button
+            variant="solid"
+            color="primary"
+            sx={{ ml: 1, height: '45px' }}
+            disabled={form.email.length === 0 || form.status === 'failure'}
```

Comment:
@hbjORbj The button should not be disabled at all times. Using native form validation is the best experience for email subscriptions in my opinion.

```js
<form onSubmit={event => ...}>
  <FormControl>
    <FormLabel>
    <Input endDecorator={<Button>} />
</form>
```